GOT-StudentVersion/GOT-StudentVersion/ViewController.swift

```diff
@@ -8,10 +8,163 @@
 
 import UIKit
 
-class ViewController: UIViewController {
+class ViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
+    
+    var episodes = GOTEpisode.allEpisodes
+    
+    
+    @IBOutlet weak var tableView: UITableView!
+    
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 7
+    }
+    
+    func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        switch section {
+        case 0:
+            return "Season One"
+        case 1:
+            return "Season Two"
+        case 2:
+            return "Season Three"
+        case 3:
+            return "Season Four"
+        case 4:
+            return "Season Five"
+        case 5:
+            return "Season Six"
+        case 6:
+            return "Season Seven"
+        default:
+            return nil
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch section {
+        case 0:
+            return GOTEpisode.seasonOne.count
+        case 1:
+            return GOTEpisode.seasonTwo.count
+        case 2:
+            return GOTEpisode.seasonThree.count
+        case 3:
+            return GOTEpisode.seasonFour.count
+        case 4:
+            return GOTEpisode.seasonFive.count
+        case 5:
+            return GOTEpisode.seasonSix.count
+        case 6:
+            return GOTEpisode.seasonSeven.count
+        default:
+            return 0
+            
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return 160
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
```

Comment:
Instead of returning every case, check what is the value of the section you're currently at and based on that give values to your cell. 

e.g.
 let episode = arrOfSeasons[indexPath.section][indexPath.row]
        var resultCell = UITableViewCell()
        if episode.season % 2 == 0 {
            guard let cell = tableView.dequeueReusableCell(withIdentifier: "GameThronesCellTwo", for: indexPath) as? CustomCell else {fatalError("GOT cell not found")}
            
            cell.GOTImage.image = UIImage(named: episode.mediumImageID)
            cell.episodeName.text = episode.name
            cell.seasonEpisode.text = "S\(episode.season) E: \(episode.number)"
            resultCell = cell
        } else if episode.season % 2 != 0 {
            guard let cell = tableView.dequeueReusableCell(withIdentifier: "GOTcell", for: indexPath) as? CustomCell else {fatalError("GOT cell not found")}
            
            cell.GOTImage.image = UIImage(named: episode.mediumImageID)
            cell.episodeName.text = episode.name
            cell.seasonEpisode.text = "S\(episode.season) E: \(episode.number)"
            resultCell = cell